src/pretix/base/models/orders.py

```diff
@@ -498,6 +500,21 @@ def __str__(self):
             return str(_("No"))
         elif self.question.type == Question.TYPE_FILE:
             return str(_("<file>"))
+        elif self.question.type == Question.TYPE_DATETIME:
+            d = dateutil.parser.parse(self.answer)
+            if self.orderposition:
+                tz = pytz.timezone(self.orderposition.order.event.settings.timezone)
+                d = d.astimezone(tz)
+            return date_format(d, "SHORT_DATETIME_FORMAT")
+        elif self.question.type == Question.TYPE_DATE:
+            d = dateutil.parser.parse(self.answer)
+            return date_format(d, "SHORT_DATE_FORMAT")
+        elif self.question.type == Question.TYPE_TIME:
+            d = dateutil.parser.parse(self.answer)
+            if self.orderposition:
+                tz = pytz.timezone(self.orderposition.order.event.settings.timezone)
+                d = d.astimezone(tz)
```

Comment:
Currently, this fails for me with the exception ``ValueError: astimezone() cannot be applied to a naive datetime`` as the TYPE_TIME values are, in contrast to the TYPE_DATETIME values, saved to the database without timezone information (just as ``18:30``, not ``18:30+00:10``). Intuitively, we would either need to

* modify TimeField or the questions form in a way that they store the timezone in the ``answer`` field with a timezone
* ensure that the ``answer`` field always contains UTC (see also my comment on the test below) and then state this here as ``d = pytz.UTC.localize(d).astimezone(tz)``

However, we have a more fundamental problem at hand here: We don't know for what day the input is! Suppose we have an event that goes from 2018-10-26 to 2018-10-28 and the organizer defines two time input questions with the texts "When do you arrive?" and "When do you leave?". The event timezone might be Europe/Berlin, but in the first case, the organizer probably wants UTC+2 and in the second case, UTC+1, since daylight saving time switches in the night of 2018-10-28. We can't ever now and asking the organizer is probably too much trouble.

So sorry for the confusion, I think we need to take all the timezone handling out again for TYPE_TIME and only keep it for TYPE_DATETIME (where it actually makes a lot of sense).

Reply:
I agree that this is a problem. How are all the datetimes defined in pretix/control represented in the database? Also, I'm wondering what timezone is used with a datetime in the one hour that is repeated when daylight savings time switches in October.

Reply:
**datetimes** are represented always as UTC in the database. On some database (I think postgres) this is explicitly stored with +00:00, on others (mysql) it is just assumed that it is always UTC.
For datetimes, that is no issue.

For **times** without a date, we don't have that up to now, and as I suggested I'd say we store them without a timezone and don't handle them at all and just output them as they are inputted by the user.

Reply:
> **datetimes** are represented always as UTC in the database.

I submitted a datetime question answer to a test event with Europe/Berlin as timezone via a local `./manage.py` sqlite instance. `str(QuestionAnswer.answer)` was `2018-10-14 14:00:00+02:00`, which is not UTC. I'm confused about that.